### PR TITLE
[Merged by Bors] - feat(MeasureTheory): weaken assumptions from `SigmaFinite` to `SFinite`

### PR DESCRIPTION
**Pull request by urkud**
- Weaken TC assumptions of many lemmas from `SigmaFinite` to `SFinite`.
- Drop some `MeasurableSet` assumptions.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/Haar/Basic.lean [hunk 1]
```diff
@@ -747,23 +747,25 @@ see the file `MeasureTheory.Measure.Haar.Unique`.
 
 variable [SecondCountableTopology G]
 
-/-- **Uniqueness of left-invariant measures**: In a second-countable locally compact group, any
+/-- **Uniqueness of left-invariant measures**: In a second-countable topological group, any
```
Comment (grunweg) [inline Mathlib/MeasureTheory/Measure/Haar/Basic.lean:750]: For the record: the current docstring was simply wrong; this is easy to confirm with the API docs.

File: Mathlib/Analysis/Convolution.lean [hunk 2]
```diff
@@ -536,15 +535,12 @@ theorem convolution_mono_right_of_nonneg {f g g' : G → ℝ}
...
     [IsAddRightInvariant μ] (h1 : f =ᵐ[μ] f') (h2 : g =ᵐ[μ] g') : f ⋆[L, μ] g = f' ⋆[L, μ] g' := by
   ext x
   apply integral_congr_ae
-  exact
-    (h1.prod_mk <|
-          h2.comp_tendsto
-            (quasiMeasurePreserving_sub_left_of_right_invariant μ x).tendsto_ae).fun_comp
-      ↿fun x y => L x y
+  exact (h1.prod_mk <| h2.comp_tendsto
+    (quasiMeasurePreserving_sub_left_of_right_invariant μ x).tendsto_ae).fun_comp ↿fun x y => L x y
```
Comment (grunweg) [inline Mathlib/Analysis/Convolution.lean:543]: While you're at it (optional):
```suggestion
    (quasiMeasurePreserving_sub_left_of_right_invariant μ x).tendsto_ae).fun_comp ↿fun x y ↦ L x y

**Review by grunweg — COMMENTED** [review]
Looks good, thank you for doing this! One optional suggestion.
maintainer merge

File: Mathlib/MeasureTheory/Measure/Haar/Basic.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/MeasureTheory/Measure/Haar/Basic.lean:750]: This should be fixed but otherwise looks good!

**Comment by mattrobball** [comment]
bors delegate+

File: Mathlib/MeasureTheory/Measure/Haar/Basic.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/MeasureTheory/Measure/Haar/Basic.lean:750]: Oh no: I meant "the previous docstring was wrong, the change in this PR fixes it". Sorry for being unclear!

Comment (sgouezel) [reply Mathlib/MeasureTheory/Measure/Haar/Basic.lean:750]: I disagree with that comment: the group is assumed to be locally compact, through the existence of a positive compact set in the statement.

Comment (mattrobball) [reply Mathlib/MeasureTheory/Measure/Haar/Basic.lean:750]: I defer to @sgouezel !

Comment (urkud) [reply Mathlib/MeasureTheory/Measure/Haar/Basic.lean]: Indeed, I shouldn't try to fix comments after midnight.

**Comment by urkud** [comment]
bors merge

**Comment by mattrobball** [comment]
Please merge again when the build succeeds

**Comment by urkud** [comment]
@mattrobball Add you can see from the logs, this is not the PR that caused the build to fail
bors merge

**Comment by mattrobball** [comment]
Sorry.

**Pull request closed without merging**